Doc/library/threading.rst

```diff
@@ -47,6 +47,65 @@ level :mod:`_thread` module.
 
 .. include:: ../includes/wasm-notavail.rst
 
+Introduction
+------------
```

Comment:
What do you think of writing this intro _before_ the pile of directives above?

Reply:
as far as I know accessibility directives should be at the top so i don't think we can move `Introduction` higher

Reply:
done. I like that now the implementation details about GIL are located next to the section about GIL